saw-remote-api/src/SAWServer/Data/Contract.hs

```diff
@@ -97,11 +114,13 @@ instance (FromJSON ty, FromJSON e) => FromJSON (Contract ty e) where
     Contract <$> o .:  "pre vars"
              <*> o .:  "pre conds"
              <*> o .:  "pre allocated"
+             <*> (maybeToList <$> o .:? "pre ghost vars")
```

Comment:
The use of `maybeToList` here confuses me. Is it always the case that there will only ever be zero or one ghost variables? If so, why not make the `preGhostPointsTos` field in `Contract` have type `Maybe (GhostPointsTo cryptolExpr)` instead of `[GhostPointsTo cryptolExpr]`?

Reply:
Yep, this was wrong. Fixed now.

Reply:
OK, it's now using `(fromMaybe [] <$> o .:? "pre ghost values")`, which I believe means that the JSON contains a `Maybe [a]`, right? What is the difference between `Just []` and `Nothing`? This would be a good thing to document.

Reply:
I think I misread this. It's not that `Just []` and `Nothing` have distinct meanings, but rather that this parses a list with a default value of `[]`. In that case, I think it would be more idiomatic to express this as:

```hs
    Contract <$> ...
             <*> ...
             <*> o .:? "pre ghost values" .!= []
             <*> ...
             <*> o .:? "post ghost values" .!= []
             <*> ...
```

Reply:
Oh, I didn't know about the `.!=` operator! That definitely makes things nicer. Fixed.